### PR TITLE
qa/multisite: enable two zonegroup yaml

### PR DESCRIPTION
**Pull request by smanjara**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by adamemerson** [comment]
Is this something to merge or are you testing something?

**Comment by smanjara** [comment]
> Is this something to merge or are you testing something?

oh just testing. adding a DNM

**Comment by smanjara** [comment]
> @smanjara this PR removes the file `two-zonegroup.yaml.disabled`, but should instead rename it back to `two-zonegroup.yaml`

oops

**Comment by smanjara** [comment]
http://qa-proxy.ceph.com/teuthology/smanjara-2024-11-20_17:01:24-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/8002132/teuthology.log

inconsistently reproducible time skew errors found:

```
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:======================================================================
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:ERROR: rgw_multi.tests.test_bucket_versioning
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:----------------------------------------------------------------------
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:Traceback (most recent call last):
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_teuthology_029bda193b1d38b67b279eb5c1037caa8408be24/virtualenv/lib/python3.10/site-packages/nose/case.py", line 170, in runTest
2024-11-20T18:52:33.760 INFO:tasks.rgw_multisite_tests:    self.test(*self.arg)
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_ceph-c_80d20f023a635a342ed280c3a62c9c6c83db1351/qa/../src/test/rgw/rgw_multi/tests.py", line 1071, in test_bucket_versioning
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:    bucket.configure_versioning(True)
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_teuthology_029bda193b1d38b67b279eb5c1037caa8408be24/virtualenv/lib/python3.10/site-packages/boto/s3/bucket.py", line 1308, in configure_versioning
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:    raise self.connection.provider.storage_response_error(
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:boto.exception.S3ResponseError: S3ResponseError: 403 Forbidden
2024-11-20T18:52:33.761 INFO:tasks.rgw_multisite_tests:<?xml version="1.0" encoding="UTF-8"?><Error><Code>RequestTimeTooSkewed</Code><Message></Message><RequestId>tx0000087bc35e38bfd5508-00673e21be-4397-b1</RequestId><HostId>4397-b1-b</HostId></Error>
```

logs on secondary zonegroup shows:

```
/plcdnh-24/?versioning
2024-11-20T17:51:58.002+0000 7f1f5e743640 10 NOTICE: request time skew too big.
2024-11-20T17:51:58.002+0000 7f1f5e743640 10 req_tp=2024-11-20T17:36:58.000000+0000, cur_tp=2024-11-20T17:51:58.002170+0000
```

there was one green run here:
https://pulpito.ceph.com/smanjara-2024-11-20_17:01:24-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/8002133/

**Comment by clwluvw** [comment]
> aws-cli/2.15.2

`test_bucket_versioning` passes for me but what is odd in your output is the user-agent. Are you sure this is coming from the rgw_multi suite?

**Comment by smanjara** [comment]
> > aws-cli/2.15.2
> 
> `test_bucket_versioning` passes for me but what is odd in your output is the user-agent. Are you sure this is coming from the rgw_multi suite? basically, 301 happens when the request is hitting the wrong zonegroup but the rgw_multi is/should use the right zone conn.
> 
> The `RequestTimeTooSkewed` we see as the original in the logs sounds like a setup issue to me rather than the code... I haven't faced such failure basically and is passing for me locally. maybe deserves a retry?

`RequestTimeTooSkewed` just comes from teuthology runs but locally when you try a simple test to create bucket and enable versioning, aws cli command throws `maximum recursion depth exceeded in comparison
` and then when you 

> `test_bucket_versioning` passes for me. basically, 301 happens when the request is hitting the wrong zonegroup. Is your local reproduce pointing to the right zone?
> 
> The `RequestTimeTooSkewed` we see as the original in the logs sounds like a setup issue to me rather than the code... I haven't faced such failure basically and is passing for me locally. maybe deserves a retry?

it has been consistently reproducible in teuthology atleast.

**Comment by clwluvw** [comment]
Ah I see the problem now... You need to backport https://github.com/ceph/ceph/pull/59305 otherwise the buckets created on zonegroup B will be created for zonegroup A and you keep getting 301 as the response.

**Comment by smanjara** [comment]
> Ah I see the problem now... You need to backport #59305 otherwise the buckets created on zonegroup B will be created for zonegroup A and you will keep getting 301 as the response and boto2 (haven't tested aws-cli) will keep retrying until it reaches the max recursion.

yeah, I used that pr too. there is a different problem there. create_bucket() path crashes on primary. will provide more details in that pr.

**Comment by clwluvw** [comment]
> create_bucket() path crashes on primary. will provide more details in that pr.

Right, I had one crash fix also here: https://github.com/ceph/ceph/pull/60254

**Comment by smanjara** [comment]
> > create_bucket() path crashes on primary. will provide more details in that pr.
> 
> Right, I had one crash fix also here: #60254

great! could you make those two commits part of a single pr? and close https://github.com/ceph/ceph/pull/59305 because running qa against it will catch this crash again.

**Comment by clwluvw** [comment]
> > > create_bucket() path crashes on primary. will provide more details in that pr.
> > 
> > 
> > Right, I had one crash fix also here: #60254
> 
> great! could you make those two commits part of a single pr? and close #59305 because running qa against it will catch this crash again.

Done, I thought for QA we can import multiple changes so we can keep these distinct to PRs.

**Comment by smanjara** [comment]
> > > > create_bucket() path crashes on primary. will provide more details in that pr.
> > > 
> > > 
> > > Right, I had one crash fix also here: #60254
> > 
> > 
> > great! could you make those two commits part of a single pr? and close #59305 because running qa against it will catch this crash again.
> 
> Done. I thought for QA we could import multiple changes so we can keep these distinct to PRs.

thanks, seena! 
with #60254 pulled in, the tests pass: https://pulpito.ceph.com/smanjara-2024-11-22_22:48:55-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/.

**Comment by clwluvw** [comment]
jenkins test api

**Comment by clwluvw** [comment]
jenkins test make check

**Comment by clwluvw** [comment]
jenkins test submodules

**Comment by clwluvw** [comment]
jenkins test make check

**Comment by clwluvw** [comment]
jenkins test submodules

**Comment by clwluvw** [comment]
jenkins test make check

**Comment by clwluvw** [comment]
> with https://github.com/ceph/ceph/pull/60254 pulled in, the tests pass: https://pulpito.ceph.com/smanjara-2024-11-22_22:48:55-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/.

Hi @cbodley, @smanjara - Can we ship this based on the QA result?

**Comment by smanjara** [comment]
> > with #60254 pulled in, the tests pass: https://pulpito.ceph.com/smanjara-2024-11-22_22:48:55-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/.
> 
> Hi @cbodley, @smanjara - Can we ship this based on the QA result?

@clwluvw yes, i'd love to! but the tests passed with #60254, #60589 and #60591. let's get those merged too.

File: qa/suites/rgw/multisite/realms/two-zonegroup.yaml
Comment (clwluvw) [inline qa/suites/rgw/multisite/realms/two-zonegroup.yaml:1]: I guess this would still leave `two-zonegroup.yaml.disabled` rather than renaming it?

**Comment by clwluvw** [comment]
> the tests passed with #60254, #60589 and #60591. let's get those merged too.

I guess this one is also needed. https://github.com/ceph/ceph/pull/59960

**Comment by smanjara** [comment]
> > the tests passed with #60254, #60589 and #60591. let's get those merged too.
> 
> I guess this one is also needed. #59960

main branch still passes without it because there are no location-constraint based tests. but that pr is also ready to be merged.

**Comment by clwluvw** [comment]
Do we need @cbodley's approval to merge this? :D

**Comment by smanjara** [comment]
@cbodley could you approve please? results here:
https://pulpito.ceph.com/smanjara-2024-11-22_22:48:55-rgw:multisite-rgw-multisite-two-zonegroups-distro-default-smithi/

I already merged the dependent prs

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
> Do we need @cbodley's approval (someone from ceph/rgw team basically) to merge this? :D

@clwluvw as a member of Ceph team, you also have the power to approve PRs for merge

the ceph/rgw team is just the list of members that get notified about every rgw pr

**Pull request merged**